### PR TITLE
Add opengraph extension

### PR DESCRIPTION
**Pull request by ItayZiv**
todo:

- [x] Licesne

- [x] Pypi Publishing

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# sphinxext-opengraph
+Sphinx extension to generate OpenGraph metadata (https://ogp.me/)
+
+## Options
+These values are placed in the conf.py of your sphinx project.
+
+### Site URL
+- *Formal Name:* `ogp_site_url`
```
Comment (Daltz333) [inline readme.md:8]: Could you reformat this to be something that shows an example configuration. Also Formal Name is confusing, what is Formal Name? You could just do `ogp_description_length: 200`

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+version = ret.stdout.decode("utf-8").strip()
+
+with open("readme.md", 'r') as readme:
+    long_description = readme.read()
+
+setuptools.setup(
+    name="sphinxext-opengraph",
+    version=version,
+    author="Itay Ziv",
+    author_email="itay220204@gmail.com",
```
Comment (Daltz333) [inline setup.py:17]: Not sure if this shows up on PyPI or not.

**Comment by Daltz333** [comment]
You can just copy the publishing script from https://github.com/Daltz333/sphinxcontrib-remoteliteralinclude/tree/master/.github/workflows

and the license should be the same as allwpilib

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import subprocess
+import setuptools
+
+# This will fail if something happens or if not in a git repository.
+# This is intentional.
+ret = subprocess.run("git describe --tags --abbrev=0", stdout=subprocess.PIPE,
+                     stderr=subprocess.PIPE, check=True, shell=True)
+version = ret.stdout.decode("utf-8").strip()
+
+with open("readme.md", 'r') as readme:
```
Comment (TheTripleV) [inline setup.py:10]:
```suggestion
with open("readme.md", 'r', encoding="utf-8") as readme:
```

**Comment by TheTripleV** [comment]
Shouldn't `__init__.py` get moved into `sphinxext/` so the installation will include it?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    long_description = readme.read()
+
+setuptools.setup(
+    name="sphinxext-opengraph",
+    version=version,
+    author="Itay Ziv",
+    author_email="itay220204@gmail.com",
+    description="Sphinx Extension to enable OGP support",
+    long_descrption=long_description,
```
Comment (TheTripleV) [inline setup.py:19]:
```suggestion
    long_description=long_description,
```

**Comment by Daltz333** [comment]
I believe it gets included either way, as remoteliteralinclude has it at the root and it works just fine

File: setup.py (hunk 3 above)
Comment (ItayZiv) [reply setup.py:10]: Is there any specific reason or is it just to be safe?

Comment (TheTripleV) [reply setup.py:10]: Just to be safe, windows doesn't default to utf-8. We ran into an issue in robotpy because of it.

**Pull request merged**